pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -380,6 +382,14 @@ func (m *managerImpl) processShutdownEvent() error {
 					}
 					status.Message = nodeShutdownMessage
 					status.Reason = nodeShutdownReason
+					if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+						podutil.UpdatePodCondition(status, &v1.PodCondition{
+							Type:    v1.AlphaNoCompatGuaranteeDisruptionTarget,
+							Status:  v1.ConditionTrue,
+							Reason:  kubelettypes.TerminatedByKubelet,
+							Message: "Shutdown manager: deleting due to node shutdown event",
```

Comment:
The pod will not be deleted, so let's rename `deleting` to `terminating` (or maybe `terminated` would be better, since the pod should be in terminal phase as part of the status update)?

Reply:
Renamed to `terminating` in the `message` field and `TerminationByKubelet` in the `reason` field. The reason being the pod condition is actually added before the pod enters the `Failed` phase due to the check for running containers: https://github.com/kubernetes/kubernetes/blob/548d135ea6b3b9a68d31c981b65e1f9e9643ea58/pkg/kubelet/status/status_manager.go#L888. The containers may not be terminated immediately as they can have a grace period.